tests/integration/source_code/test_jobs.py

```diff
@@ -118,6 +125,27 @@ def test_workflow_linter_lints_job_with_import_pypi_library(
     assert len([problem for problem in problems if problem.message == "Could not locate import: pytest"]) == 0
 
 
+def test_lint_local_code(simple_ctx):
+    # no need to connect
+    light_ctx = simple_ctx.replace(languages=Languages(MigrationIndex([])))
+    ucx_path = Path(__file__).parent.parent.parent.parent
+    path_to_scan = Path(ucx_path, "src")
+    linter = LocalCodeLinter(
+        light_ctx.file_loader,
+        light_ctx.folder_loader,
+        light_ctx.path_lookup,
+        light_ctx.dependency_resolver,
+        lambda: light_ctx.languages,
+    )
+    old_stdout = sys.stdout
+    try:
+        sys.stdout = StringIO()
```

Comment:
This is incorrect, as we run unit tests in multiple threads. It'll get flaky

Don't mutate global shared state like stdout, inject it to a method instead if non-default.

Reply:
done